osu.Server.Queues.ScoreStatisticsProcessor/Commands/Queue/WatchHighScoresCommand.cs

```diff
@@ -0,0 +1,237 @@
+                    Console.WriteLine($"Continuing watch mode from last processed legacy score (score_id: {entry.score_id} queue_id: {entry.queue_id})");
+                }
+                else
+                {
+                    // There may have been no new scores since the last run.
+                    entry = db.QuerySingle<ScoreProcessQueue>($"SELECT * FROM score_process_queue WHERE mode = {RulesetId} AND score_id = {lastImportedLegacyScoreId}");
+                    lastQueueId = entry.queue_id;
+                    Console.WriteLine($"Continuing watch mode from last processed legacy score (queue_id: {entry.queue_id})");
+                }
+            }
+            else
+            {
+                lastQueueId = firstEntry.queue_id;
+                Console.WriteLine($"WARNING: Continuing watch mode from start of the processing queue (score_id: {firstEntry.score_id} queue_id: {firstEntry.queue_id})");
+                Console.WriteLine("This implies that a full import hasn't been run yet, you might want to run another import first to catch up.");
+                Console.WriteLine();
+                Console.WriteLine("You have 10 seconds to decide if you really want to do this..");
+                await Task.Delay(10000, cancellationToken);
+            }
+
+            if (SkipScoreProcessor)
+            {
+                elasticQueueProcessor = new ElasticQueuePusher();
+                Console.WriteLine($"Indexing to elasticsearch queue(s) {elasticQueueProcessor.ActiveQueues}");
+            }
+            else
+            {
+                scoreStatisticsQueueProcessor = new ScoreStatisticsQueueProcessor();
+                Console.WriteLine($"Pushing imported scores to redis queue {scoreStatisticsQueueProcessor.QueueName}");
+            }
+
+            if (DryRun)
+                Console.WriteLine("RUNNING IN DRY RUN MODE.");
+
+            Console.WriteLine();
+            Console.WriteLine("Starting processing in 5 seconds...");
+            await Task.Delay(5000, cancellationToken);
+
+            using (var dbMainQuery = DatabaseAccess.GetConnection())
+            {
+                while (!cancellationToken.IsCancellationRequested)
+                {
+                    HighScore[] highScores = (await dbMainQuery.QueryAsync<HighScore>($"SELECT * FROM osu.score_process_queue LEFT JOIN {highScoreTable} h USING (score_id) WHERE queue_id >= @lastQueueId AND mode = {RulesetId} ORDER BY queue_id LIMIT 1000", new
+                                             {
+                                                 lastQueueId,
+                                                 RulesetId,
+                                             }))
+                                             // there might be multiple queue entries for the same insert. this can cause issues due to how we do the mapping lookup so let's fix that.
+                                             .DistinctBy(s => s.score_id)
```

Comment:
The thing that worried me slightly here is I'm not sure _which_ of the duplicate queue entries this `.DistinctBy()` is going to pick. But I'm not entirely sure this matters either, because the worst case scenario of what could happen here is:

- select returns 2 rows for single score, e.g. one with `queue_id = 100000` and `queue_id = 100002`
- distinct picks one of the two arbitrarily
- the `queue_id` is later only read from the last score in a batch, to determine where to start the next batch
- so the worst case scenario here would be basically that some scores will be imported twice (because batches will partially overlap)

so I *think* it still works......